### PR TITLE
Add pipeline performance trends

### PR DESCRIPTION
**Pull request by kassenq**
### What does this PR do?
We recently shipped changes to the pipeline branch overview page and the Pipeline Executions / explorer pages, and this PR updates the Pipeline Visibility index/overview page to include these features.

### Motivation
See above

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by alai97 — APPROVED** [review]
Looks good!

File: content/en/continuous_integration/pipelines/_index.md [hunk 1]
```diff
@@ -45,6 +45,14 @@ Click into a specific pipeline to see the _Pipeline Details_ page which provides
...
 Get insights on the selected pipeline such as total and failed executions over time, build duration percentiles, and total time spent breakdown by stage. There are also summary tables for stages and jobs so you can quickly sort them in terms of duration, percentage of overall execution time, or failure rate.
 
+When viewing the _default_ branch for a pipeline, each stage and job summary table on the page contains a _Duration Change_ graph that visualizes the absolute and relative change in duration of stages and jobs, along with a graph of _Errored Executions_.
+
+{{< img src="ci/pipeline_job_summary_duration_change.png" alt="Job summary table" style="width:100%;">}}
+
+When viewing a feature branch for a pipeline, there are two graphs for _Errored Executions_: one for failed executions in this feature branch, and another for comparing failures in this branch to other failures. You have the option to toggle between `Other Branches` and `Specific Branch`, so you can compare failed executions in this feature branch with aggregated failures across all other branches, or directly compare failed executions in this feature branch with failed executions in another branch of your choice.
```
Comment (KickingTheTV) [inline content/en/continuous_integration/pipelines/_index.md:52]: I would just add a sentence or two here about why this comparison is useful against "Other Branches" and "Specific Branch" - why the customer would use each basically

**Review by KickingTheTV — CHANGES_REQUESTED** [review]
Overall LGTM - would just update this one item with more information about why it's useful

File: content/en/continuous_integration/pipelines/_index.md (hunk 1 above)
Comment (kassenq) [reply content/en/continuous_integration/pipelines/_index.md]: Makes sense!

File: content/en/continuous_integration/pipelines/_index.md [hunk 2]
```diff
@@ -45,6 +45,16 @@ Click into a specific pipeline to see the _Pipeline Details_ page which provides
...
+
+{{< img src="ci/pipeline_job_summary_duration_change.png" alt="Job summary table" style="width:100%;">}}
+
+When viewing a feature branch for a pipeline, there are two graphs for _Errored Executions_: one for failed executions in this feature branch, and another for comparing failures in this branch to other failures. You have the option to toggle between `Other Branches` and `Specific Branch`. Selecting `Other Branches` compares failed executions in this feature branch with aggregated failures across all other branches to help you distinguish between introduced (local) execution failures and inherited (global) execution failures. Toggling to `Specific Branch` directly compares failed executions in this feature branch with failed executions in another branch of your choice to help you specifically track changes in pipeline performance.
+
+or directly compare failed executions in this feature branch with failed executions in another branch of your choice.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/_index.md:54]:
```suggestion
Optionally, you can directly compare failed executions in this feature branch with failed executions in another branch of your choice.
```

**Review by KickingTheTV — APPROVED** [review]

**Pull request merged**